### ISSUE TITLE
Ability to determine Legend sort ordering

### ISSUE THREAD

**Issue by YeahButHats** [dc-js/dc.js#1121]
Most of the time, the order of the legend doesn't matter, but then you have situations like this:

![image](https://cloud.githubusercontent.com/assets/1030358/14020569/1d6b428a-f1ae-11e5-879a-1b69613f11d5.png)

In these cases (when you don't want alphabetical), it would be nice to be able to specify the order of the legend.

**Comment by gordonwoodhull** [comment]
Agreed, this would be a nice feature to have. If you want a workaround for now, it should be pretty easy to override `chart.legendables()` to change the order.

Something like:

```js
    dc.override(_chart, 'legendables', function () {
        var legendables = _chart._legendables();
        return legendables.sort(function(a, b) { /* your sort criteria here */ });
    })
```

**Comment by gordonwoodhull** [comment]
Thanks for the contribution, @YeahButHats.

**Comment by jaklub** [comment]
Had a similar issue but with an area/stacked line chart. The line chart already overides legendables (line-chart.js 488 ) so the workaround above at first resulted in legendables() calling it self and getting stuck in a loop. 

But by "resetting" the first overide call and then combining the workaround above with the line chart legendables function, i got something which seem to work for me.
 
```javascript
dc.override(chart, 'legendables', chart._legendables);

dc.override(chart, 'legendables', function () {
    var legendables = this._legendables();
    if (!this.dashStyle()) {
        return legendables.reverse();
    }
    return legendables.map(function (l) {
        l.dashstyle = this.dashStyle();
        return l.reverse();
    });
});
```

**Comment by grugknuckle** [comment]
Is there documentation on the legendables property? I can't seem to find it anywhere in [the documentation for dc.js ver 2.1.9](https://github.com/dc-js/dc.js/blob/master/web/docs/api-latest.md#dc.barChart).

**Comment by gordonwoodhull** [comment]
@grugknuckle, that's a good point, it should be documented.

I think it was not originally because it's an interface between charts and the legend, not really a user-level API. But it can be very useful to override it.

Very few of the internal APIs are documented.

**Comment by grugknuckle** [comment]
I only found out about it from your S.O. post here ...

https://stackoverflow.com/questions/48336596/how-to-add-legend-for-a-bar-chart-with-different-colors-in-dc-js
